src/BackupMetadata.cs

```diff
@@ -2,39 +2,39 @@ namespace AgDatabaseMove
 {
   using System;
   using System.Collections.Generic;
+  using System.Linq;
   using SmoFacade;
 
-  public class BackupMetadataEqualityComparer : IEqualityComparer<BackupMetadata>
+  public class StripedBackupEqualityComparer : IEqualityComparer<BackupMetadata>
   {
-    /// <summary>
-    /// This is used for checking similar backups (like striped backups)
-    /// </summary>
-    /// <returns>bool</returns>
-    public bool EqualsExceptForPhysicalDeviceName(BackupMetadata x, BackupMetadata y)
+    private static StripedBackupEqualityComparer _instance = null;
+    private StripedBackupEqualityComparer() { }
+
+    public static StripedBackupEqualityComparer Instance {
+      get
+      {
+        if (_instance == null)
+        {
+          _instance = new StripedBackupEqualityComparer();
+        }
+        return _instance;

```

Comment:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/null-coalescing-assignment

I think this can be simplified using `??=`. Something like
```
_instance ??= new StripedBackupEqualityComparer();
return _instance ;
```

or even 
` return _instance ??= new StripedBackupEqualityComparer();`

Reply:
yeah... this doesn't even need to be a public property with a `Getter`

```
private static readonly Lazy<StripedBackupEqualityComparer> s_instance = new Lazy<StripedBackupEqualityComparer>();

    private StripedBackupEqualityComparer() { }

    public static StripedBackupEqualityComparer Instance => s_instance.Value;
```

Reply:
ah cool, didn't know you could make anything a lazy initialisation - will try it out
https://docs.microsoft.com/en-us/dotnet/api/system.lazy-1?view=net-6.0